### DISCUSSION TITLE
Next stable release on docker?

### DISCUSSION THREAD

**Discussion by EVaisman** [getredash/redash#6647] [General]
I see that the docker image with the latest tag has been updated recently. We currently point the redash docker image to the 10.1.0 tag (which is marked as the latest in the [releases page](https://github.com/getredash/redash/releases)). Updating it to point to lastest tag is a bit risky for us and we'd like the version to be explicitly specified. Is it possible to update the latest fixed version?

**Comment by justinclift** [comment]
The next production suitable release is likely to be available early in the new year. :smile:

The `preview` tag isn't really suitable for production use yet, so you're probably best to stick with the older 10.1.0 release for now. :sweat_smile:

**Reply by M-arghandiwal** [reply]
Is there an exact date?

**Reply by justinclift** [reply]
Nope, not yet.

**Reply by rodrigost23** [reply]
It's almost March already. Is there a roadmap, a list of issues that need to be fixed before release or something?

**Comment by eradman** [comment]
I proposed a change that I think can help us make progress:

_Publish preview Docker image when release candidate is tagged_: https://github.com/getredash/redash/pull/6787

We can see that many people have been using the preview image, which is great, but bug reports are hard to follow because each person may be testing a different snapshot. With this change we should be connect bug reports and confirmation of fixes with specific tags:

24.01.0-dev
24.02.0-dev
24.03.0-dev
...

In integration tests this also makes the previous and next version predictable, so we can step through each release without having to pin an image to individual SHA1s

```diff
--- a/redash-email/docker-compose.yml
+++ b/redash-email/docker-compose.yml
@@ -2,7 +2,7 @@
 # For a production example please refer to getredash/setup repository on GitHub.
 version: "2.2"
 x-redash-service: &redash-service
-  image: redash/redash:preview
+  image: redash/preview:24.02.0-dev
   env_file:
     - .env
```

**Comment by rodrigost23** [comment]
It's already June, and we're stuck with EOL versions of Python, boto3, and so on, despite the fact that there isn't yet a stable version that's not at least three years old.

I'm not asking for new features, just for a version with up-to-date dependencies that is considered stable...

**Reply by zachliu** [reply]
The `24.06.0-dev` released on 05/31/2024 works fine if you are in a pinch.
It got rid of those requirements.txt so that your build is a breeze. No need to worry about upstream library releases break your deployment or cause some runtime issues.

**Reply by phillipjohnson** [reply]
@zachliu @justinclift If `24.06.0-dev` is stable, is it possible to tag this version in Docker Hub so we can start testing a stable release in our dev environment?

**Reply by justinclift** [reply]
@phillipjohnson There's still an outstanding serious bug with embeds that I've seen break when Redash is upgraded from v10.1.x.

Personally I'm currently doing the tail end of a large data centre migration, and will look at the above bug once that's finished.  Hopefully that'll be around the end of this week, or start of next week.

**Reply by lucydodo** [reply]
@phillipjohnson I'm not sure if I understand you correctly, but are you referring to pull that image from Docker Hub?
If so, you can pull it with the following command: `$ docker pull redash/preview:24.06.0-dev`

**Reply by phillipjohnson** [reply]
@lucydodo Ah, thanks! I was just looking in the main repo.

**Reply by zachliu** [reply]
@justinclift which bug are you referring to? i deployed `24.06.0-dev` into a production environment. maybe i can help

**Reply by justinclift** [reply]
@zachliu I did a test upgrade of the Redash server about two months ago that these embeds are generated by:

https://sqlitebrowser.org/stats/

After the upgrade, everything in the server seemed happy (Queries, Dashboards, etc) **except** for all three embeds on that page.  Instead it gave an error:

![image](https://github.com/getredash/redash/assets/406299/c0a2866c-c92e-4849-8e01-507830ec7265)


No amount of refreshing the query (even manually), re-running it, etc helped.

Tried it again about 2 weeks ago, except I loaded a full database dump into a new PostgreSQL instance then upgraded.  Exactly the same behaviour.

I'm super focused on getting the last parts of a data centre migration finished, so I haven't had any spare cycles to look at the problem.  It might be something totally simple... or not, completely unknown so far.

**Reply by zachliu** [reply]
@justinclift omg yes yes, i saw ~the same behavior~ similar behavior and thought something went wrong with my db migration too!!! i'm not using embeds but somehow ppl couldn't download their query results. when they go to the "download url" they got an empty page `{"message": "No cached result found for this query."}`

But in my case, i told everyone to re-run their queries manually so that the "cache" got the results :thinking:

**Reply by justinclift** [reply]
Wish that worked for me when I tried it. :wink:

**Reply by rcheatham-q** [reply]
I also would love to see a stable version released soon

**Reply by TheRaLabs** [reply]
i am begging you to have a stable release

**Reply by justinclift** [reply]
@TheRaLabs Give it a few weeks.  I'm personally switching to focus full time on Redash development next week, so will be able to get the last remaining issues (like the above) fixed then.

**Comment by eradman** [comment]
I started a wiki page for [Release Notes](https://github.com/getredash/redash/wiki/Release-Notes) based on the -dev snapshots

@EVaisman @bear0330 @moontalk @nhayato @dtaniwaki @tsawada @rodrigost23 @kaelaela please take some time to update this page

**Comment by zachliu** [comment]
you guys are truly great! and i mean it!
when pestering opensource maintainers for new releases, the responses i usually got was `you didn't pay for it, so shut up and wait` :joy:

**Reply by justinclift** [reply]
Heh Heh Heh.  Yeah, some projects have a less-than-friendly Community. :frowning:

**Comment by phillipjohnson** [comment]
Bumping this to see if we have any update on a stable release? The release notes and tags for the DEV releases are helpful!

**Reply by phillipjohnson** [reply]
@eradman @justinclift Pinging directly to see if there is an update, thanks!

**Reply by justinclift** [reply]
The currently monthly release is pretty decent.  I'm having to sort out some personal stuff currently though, so probably won't be looking at Redash with serious effort until the new year. :frowning:

**Reply by eradman** [reply]
I have deployed 24.11 widely and have had _zero_ issues. The only thing I would so is apply to this branch is https://github.com/getredash/redash/pull/7240 to a avoid a spurious warning.

I'd say we just need a ticket titled "v24.11 Release Plan"! @phillipjohnson does it make sense to you to target 24.11, 24.12 or wait for 25.01 to be tagged?

**Reply by phillipjohnson** [reply]
@eradman That's awesome! No preference on which version is best to tag as a release. If there is a list of items that need to be completed before an official release, I'd be happy to take a look and see where we can help out. Other than adjustments to the docker-compose file, if there are any manual steps required for an upgrade, that would be helpful. cc: @zachliu who had success with an upgrade based on [comment below.](https://github.com/getredash/redash/discussions/6647#discussioncomment-11159086)

**Reply by eradman** [reply]
Next we should review the release checklist template to see if there are any steps we should add

https://github.com/getredash/redash/wiki/New-version-release-checklist

**Reply by eradman** [reply]
A regression was  reported in https://github.com/getredash/redash/issues/7247, I'm looking into this before starting a release ticket.

**Comment by mjgp2** [comment]
I'm going to be upgrading a v10 to v24 (when it's released).

Can I offer any help writing any migration docs, or are they somewhere already that I haven't found?

Thanks :)

**Reply by zachliu** [reply]
maybe a new `migration` section at https://redash.io/help (https://github.com/getredash/website)?

i want to brag about meself going from `9.0.0-beta` to `24.10.0-dev` :muscle:
i'd say i ironed out most if not all bumps already, with https://github.com/getredash/redash/issues/6954 being the most significant one

**Reply by eradman** [reply]
> maybe a new `migration` section at https://redash.io/help (https://github.com/getredash/website)?

There is a how-to-upgrade page, which just needs to be updated

https://redash.io/help/open-source/admin-guide/how-to-upgrade/

**Comment by eradman** [comment]
`redash/redash:25.1.0` has been published! https://hub.docker.com/r/redash/redash/

Following up on checklist items in https://github.com/getredash/redash/issues/7251
